src/website/webview/views.py

```diff
@@ -401,3 +403,44 @@ class NixpkgsIssueListView(ListView):
 
     def get_queryset(self) -> BaseManager[NixpkgsIssue]:
         return NixpkgsIssue.objects.all()
+
+
+class NixderivationPerChannelView(ListView):
+    template_name = "affected_list.html"
+    context_object_name = "affected_list"
+    paginate_by = 4
+
+    def get_queryset(self) -> Any:
+        channel_filter_value = self.kwargs["channel"]
+        channel = get_object_or_404(NixChannel, channel_branch=channel_filter_value)
+
+        return (
+            NixpkgsIssue.objects.prefetch_related(
+                "cve", "derivations", "derivations__parent_evaluation"
+            )
+            .values(issue_id=F("id"), issue_code=F("code"), issue_status=F("status"))
+            .filter(issue_status=IssueStatus.AFFECTED)
+            .annotate(
+                cve_id=F("cve__id"),
+                cve_code=F("cve__cve_id"),
+                cve_state=F("cve__state"),
+                drv_id=F("derivations__id"),
+                drv_name=F("derivations__name"),
+                drv_system=F("derivations__system"),
+                drv_path=F("derivations__derivation_path"),
+                channel_id=F("derivations__parent_evaluation__channel_id"),
+            )
+            .filter(channel_id=channel.channel_branch)
+        )
+
+    def get_context_data(self, **kwargs: Any) -> Any:
+        context = super().get_context_data(**kwargs)
+        context["channels"] = ["NIXOS-UNSTABLE", "NIXOS-24.05", "NIXOS-23.11"]
```

Comment:
Why are existing channels hard coded? I didn't take time to fully penetrate the code, but what speaks against taking the list of channels from the evaluation data?

Reply:
Implemented in the last commit. The tabs look like this now:
![image](https://github.com/user-attachments/assets/b71f9445-345c-4387-baa5-93dbd7d85c8b)